src/builder/create_attachment.rs

```diff
@@ -251,22 +319,19 @@ impl<'a> EditAttachments<'a> {
     }
 
     /// Clones all new attachments into a new Vec, keeping only data and filename, because those
-    /// are needed for the multipart form data. The data is taken out of `self` in the process, so
-    /// this method can only be called once.
+    /// are needed for the multipart form data.
     #[cfg(feature = "http")]
-    pub(crate) fn take_files(&mut self) -> Vec<CreateAttachment<'a>> {
-        let mut files = Vec::new();
-        for attachment in &mut self.new_and_existing_attachments {
-            if let NewOrExisting::New(attachment) = attachment {
-                let cloned_attachment = CreateAttachment::bytes(
-                    std::mem::take(&mut attachment.data),
-                    attachment.filename.clone(),
-                );
-
-                files.push(cloned_attachment);
-            }
-        }
-        files
+    pub(crate) fn new_attachments(&self) -> Vec<CreateAttachment<'a>> {
```

Comment:
Can this take `self`, Vec::retain for the new attachments, then return that `Vec`? I don't see where it is being used without ownership.

Reply:
Taking `self` wouldn't be possible, as the builder gets passed to be serialized as part of the payload, and that actually requires that it's the _existing_ attachments that get retained inside the builder, whereas the new attachments are passed as part of the multipart upload. I accidentally got rid of that behavior so I should revert it.